src/app/modules/main/chat_section/module.nim

```diff
@@ -792,36 +803,35 @@ method onCommunityTokenPermissionDeleted*(self: Module, communityId: string, per
   singletonInstance.globalEvents.showCommunityTokenPermissionDeletedNotification(communityId, "Community permission deleted", "A token permission has been removed")
 
 method onCommunityTokenPermissionCreated*(self: Module, communityId: string, tokenPermission: CommunityTokenPermissionDto) =
-  if tokenPermission.`type` == TokenPermissionType.BecomeMember:
-    let tokenPermissionItem = self.buildTokenPermissionItem(tokenPermission)
+  let tokenPermissionItem = self.buildTokenPermissionItem(tokenPermission)
+  if tokenPermissionItem.tokenCriteriaMet:
+    self.view.setAllTokenRequirementsMet(true)
+  self.view.tokenPermissionsModel.addItem(tokenPermissionItem)
+  self.view.setRequiresTokenPermissionToJoin(true)
 
-    if tokenPermissionItem.tokenCriteriaMet:
-      self.view.setAllTokenRequirementsMet(true)
-
-    self.view.tokenPermissionsModel.addItem(tokenPermissionItem)
-    self.view.setRequiresTokenPermissionToJoin(true)
   singletonInstance.globalEvents.showCommunityTokenPermissionCreatedNotification(communityId, "Community permission created", "A token permission has been added")
 
 method onCommunityTokenPermissionUpdated*(self: Module, communityId: string, tokenPermission: CommunityTokenPermissionDto) =
```

Comment:
These functions make me realize that these checks should be done in status-go (they probably are already done there), and the signal should contain the result already (if the condition is met). That way we wouldn't have to calculate it again in Nim.

Not something to be done now, but if I'm right, then later that's something we should do

Reply:
This is what's in the works. In fact, we already have `CheckPermissionsToJoinCommunity()` that we can call which will give us all we need.

We'll extend this API now with additional info, but for doing checks just for the UI, this is enough.

So yes, we can move away from relying on walletService and collectibles service and simply call that API instead.

Reply:
However, we can probably do that in a follow up PR

Reply:
> However, we can probably do that in a follow up PR

Yes 100%. I was just pointing it out, but I glad it's being considered and done. Very cool.